haskell/test/SMT/BaseDAO/Common.hs

```diff
@@ -0,0 +1,20 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module SMT.BaseDAO.Common
+ ( authorizeAdmin
+ ) where
+
+import Universum
+
+import Lorentz hiding (not, cast, checkSignature)
+import Tezos.Address (mkKeyAddress)
+import Tezos.Crypto (checkSignature)
+
+import Ligo.BaseDAO.Types
+import SMT.Types
+
+authorizeAdmin :: ContractCall Parameter -> Storage -> Either ModelError Storage
+authorizeAdmin cc store =
+  if (cc & ccSender) == (store & sAdmin) then Right store
+  else Left NOT_ADMIN
```

Comment:
Here's an example of how the `ModelMonad` makes these functions cleaner, this would be:
```haskell
authorizeAdmin :: ContractCall Parameter -> ModelMonad
authorizeAdmin cc store = do
  modelState <- ask
  when (ccSender cc /= (sAdmin $ csStorage modelState)) $ throwE NOT_ADMIN
```

Also, one more suggestion is to move this and similar model-related modules to a name containing `Model`, e.g. for this one `SMT.Model.Common` I think would be better, as it states more clearly that's about the Haskell implementation

Reply:
I've renamed them to to use the name `Model` instead.